test/services/DexPoolScanner.test.ts

```diff
@@ -0,0 +1,264 @@
+    mockedMappingRepository = sinon.createStubInstance(MappingRepository);
+    mockedUniswapV3Factory = sinon.createStubInstance(UniswapV3Factory);
+    mockedDexRepository = sinon.createStubInstance(DexRepository);
+
+    mockedUniswapV3Factory.getPoolCreatedEvents.resolves([
+      {token0: '0xABCD', token1: '0xABFD', fee: BigInt(400.0), pool: '0xPOLKD', anchor: 123},
+    ]);
+
+    mockedMappingRepository.set.resolves();
+    mockedStaticJsonRpcProvider.getBlockNumber.resolves(50_000);
+
+    sinon.stub(DexProtocolFactory, 'create').callsFake(() => mockedUniswapV3Factory);
+    sinon.createStubInstance(DexPoolScannerAgent);
+
+    mockedBlockchainProviderRepository.get.returns(mockedStaticJsonRpcProvider);
+
+    mockedApplicationGet = sinon.stub(Application, 'get');
+    mockedApplicationGet.onCall(0).returns(mockedBlockchainProviderRepository);
+    mockedApplicationGet.onCall(1).returns(mockedMappingRepository);
+    mockedApplicationGet.onCall(2).returns(mockedDexRepository);
+    mockedApplicationGet.onCall(3).returns(settings);
+    mockedApplicationGet.onCall(4).returns(mockedLogger);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('#constructor', () => {
+    describe('when settings are invalid', () => {
+      describe('when missing providerURL', () => {
+        beforeEach(() => {
+          settings = {
+            blockchains: {
+              ethereum: {
+                providerUrl: [],
+              },
+            },
+            dexes: {
+              [ChainsIds.ETH]: {
+                [DexProtocolName.UNISWAP_V3]: {
+                  active: true,
+                },
+              },
+            },
+          } as unknown as Settings;
+
+          mockedApplicationGet.onCall(3).returns(settings);
+        });
+
+        it('thrown error', async () => {
+          expect(() => new DexPoolScanner(ChainsIds.ETH, DexProtocolName.UNISWAP_V3)).to.throw(
+            Error,
+            '[DexPoolScanner][ethereum][uniswapV3] provider URL not set',
+          );
+        });
+      });
+
+      describe('when startBlock is 0', () => {
+        beforeEach(() => {
+          settings = {
+            blockchains: {
+              ethereum: {
+                providerUrl: ['test'],
+              },
+            },
+            dexes: {
+              [ChainsIds.ETH]: {
+                [DexProtocolName.UNISWAP_V3]: {
+                  active: true,
+                  startBlock: 0,
+                },
+              },
+            },
+          } as unknown as Settings;
+
+          mockedApplicationGet.onCall(3).returns(settings);
+        });
+
+        it('thrown error', async () => {
+          expect(() => new DexPoolScanner(ChainsIds.ETH, DexProtocolName.UNISWAP_V3)).to.throw(
+            Error,
+            '[DexPoolScanner][ethereum][uniswapV3] startBlock must be higher than 0',
+          );
+        });
+      });
+    });
+  });
+
+  describe('#apply', () => {
```

Comment:
These ones look more like integration tests rather than unit tests. I would prefer to call the `apply(...)` method directly and not via the `run(...)` method.

Reply:
apply method is private, so I renamed the describe to "run".
So not sure if it is integration, since I mocked the DB and Contract calls